design/open/metadata_kv/index.md

```diff
@@ -265,6 +265,28 @@ This takes time proportional to the amount of changes to be committed and is unb
 
 With the optimistic version, readers and writers end up retrying in during exactly 2 "constant time" operations during a commit: during the initial update with a new `staging_token`, and again when clearing `sealed_tokens`.
 The duration of these operations is of course, not constant - it is (well, should be) very short, and not proportional to the size of the commit in any way.
+
+### Partitioning
+
+Partitioning the KV key space designed to increase performance. The correctness of the KV interface without partitioning remains.
+We rely on 2 assumptions for adding the partitioning:
+1. lakeFS keys can be partitioned in a manner that doesn't require access to 2 partitions in a single operation.
+The only operation of the API that handles more than a single key is `Scan` and there's no use-case for scanning more than a single partition.
+1. Performance decreases if the keys are not partitioned. For example, a transaction on a kv Postgres table will lock the entire table
+during the execution. Working on partitioned key-space will only block the single partition.
+
+The KV store implementation is in charge of managing the partitioned storage. The user the KV store
+is responsible for choosing the appropriate partitions. For example, the number of keys used for authentication
+& authorization is relatively small, so using the same 'auth' partition for all auth entities is appropriate.
+However, the number of entries (keys) under a single staging token is possibly the size of the entire repository,
+so creating a partition for each staging token is the right strategy.
+
+Guidelines:
+- Partition keys are also a namespace for the key. i.e. the combination of (partitionKey,key) is unique in the KV database,
+but (key) is not guaranteed to be unique.
+- Although possible in some implementations, the interface will not support `dropPartition`. 
```

Comment:
Not required... but it will speed up commits on some KVs.  Obviously on PostgreSQL. Butalso on DynamoDB because now you can drop partitions by iterating over keys in a goroutine after the commit is done and not have to worry about iterator invalidation. If there are no single-key deletes, you can Scan 1000 keys at a time in your iterator and give correct results. This speedup due to allowable batching should work with most KVs. 

Iterators become faster this way.

Reply:
Thx, added this clarification.